modules/openapi-generator/src/main/resources/python/python-experimental/api_client.mustache

```diff
@@ -542,3 +561,180 @@ class ApiClient(object):
                     raise ApiValueError(
                         'Authentication token must be in `query` or `header`'
                     )
+
+{{#hasHttpSignatureMethods}}
+    def get_http_signature_headers(self, resource_path, method, headers, body, query_params):
+        """
+        Create a cryptographic message signature for the HTTP request and add the signed headers.
+        
+        :param resource_path : resource path which is the api being called upon.
+        :param method: the HTTP request method.
+        :param headers: the request headers.
+        :param body: body passed in the http request.
+        :param query_params: query parameters used by the API.
+        :return: instance of digest object 
```

Comment:
Can you add type descriptions to these params?

Reply:
yes, doing.